### DISCUSSION TITLE
Can a single ImagePolicy be used for multiple ImageRepository

### DISCUSSION THREAD

**Discussion by SonalinP** [fluxcd/flux2#3588] [Q&A]
I have multiple container images for my apllications stored in a docker hub (OCI) and helm charts (umbrellas chart) store in  helm repository .
I want flux to detect the new images and update the image details in helm charts accordingly.

I need help on the below queries :

1) Is there any way to use single ImageRepository for different images like GitRepository /HelmRepository
2) If multiple ImageRepository is needed in this case , is it possible to use a single ImagePolicy where we can map different imageRepositoryRef ?
3) How can I achieve helm charts (which are present as helm package in helm repo) to be updated with latest image by flux ?

**Comment by stefanprodan** [comment]
> How can I achieve helm charts (which are present as helm package in helm repo) to be updated with latest image by flux ?

There is no support for this type of automation. You need to push the Helm charts to a container registry to use Flux automation, please see:
- https://fluxcd.io/flux/cheatsheets/oci-artifacts/#helm-oci
- https://fluxcd.io/flux/cheatsheets/oci-artifacts/#helm-oci-automation

**Reply by SonalinP** [reply]
Thanks for your reply .
Do we need to keep charts under container registry only ? Because I tried with charts which are stored in git repo and flux is able to update latest image in helm charts .

Could  you please help with below queries  
1) Is there any way to use single ImageRepository for different container  images like we use single GitRepository /HelmRepository
2) If multiple ImageRepository is needed in this case , is it possible to use a single ImagePolicy where we can map different imageRepositoryRef ?

**Reply by stefanprodan** [reply]
The answer to both questions is no. But you don't need to look at container images anymore if your charts are stored in a container registry. Instead of patching the container images, Flux will patch the chart version which contains the container right images and tags.

**Comment by SonalinP** [comment]
I have already checked the flux documentation for OCI .
But still steps are not clear to me .
I have helm charts where in deployment.yaml I have the image : <which is pushed to container registry >.
Now if I push the charts to same docker hub ,how flux will patch the chart without checking for images .

Could you please explain the flow by giving  sample example for better understanding.
Currently we dont have CI in place , so we are pushing the container image manually .
